src/scripts/components/widgets/modals/ModalChallenge.js

```diff
@@ -0,0 +1,162 @@
+/* @flow */
+import paratii from 'utils/ParatiiLib'
+import React, { Component } from 'react'
+import styled from 'styled-components'
+import Text, { Strong } from 'components/foundations/Text'
+import Button from 'components/foundations/Button'
+import UserRecord from 'records/UserRecords'
+import { ModalContentWrapper, ModalScrollContent } from './Modal'
+
+type Props = {
+  id: string,
+  title: string,
+  description: string,
+  author: string,
+  user: UserRecord,
+  selectedVideoId: string,
+  closeModal: () => void,
+  notification: (Object, string) => void,
+  loadBalances: () => void
+}
+
+const Title = styled.h2`
+  color: ${props => props.theme.colors.Modal.title};
+  font-size: ${props => props.theme.fonts.modal.title};
+  margin-bottom: 25px;
+`
+
+const Highlight = styled(Text)`
+  color: ${props => props.theme.colors.Modal.hightlight};
+  margin-bottom: 14px;
+`
+
+const MainText = styled(Text)`
+  margin-bottom: 35px;
+`
+
+const Anchor = Button.withComponent('a')
+
+const Footer = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  margin-top: 50px;
+  width: 100%;
+`
+
+class ModalChallenge extends Component<Props, Object> {
+  onSubmit: (e: Object) => void
+  disableButton: (e: Object) => void
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      errorMessage: false,
+      agreedTOC: false, // TODO,
+      stakeAmount: 0,
+      isChallenging: false
+    }
+    this.onSubmit = this.onSubmit.bind(this)
+    this.disableButton = this.disableButton.bind(this)
+  }
+
+  disableButton = () => {
+    return this.state.isChallenging
+  }
+
+  async onSubmit (event: Object) {
+    event.preventDefault()
+    this.props.notification({ title: 'Processing the challenge...' }, 'warning')
+    try {
+      await paratii.eth.tcr.approveAndStartChallenge(this.props.selectedVideoId)
+    } catch (e) {
+      this.props.notification({ title: e.message }, 'error')
```

Comment:
@jellegerbrandy is it possible to enumerate all of the possible error messages that this operation could return so that we can write translations for them?